### PR TITLE
Extend Lossy Arrays to Appropriately Handle Decodables instead of just LosslessDecodables

### PR DESCRIPTION
**Pull request by matolah**

### CODE REVIEW

File: Sources/Kodable/Codable+Lossless.swift [hunk 1]
```diff
@@ -38,14 +38,35 @@ struct LosslessValue<T: LosslessDecodable>: Decodable {
...
+        }
+    }
+
+    var elements: [Element]
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        let wrappers = try container.decode([ElementWrapper].self)
+        self.elements = wrappers.compactMap(\.element)
+    }
```
Comment (rogerluan) [inline Sources/Kodable/Codable+Lossless.swift]: I'd add 1 extra test case, which would be the exact case that we saw a failure on:

Try to parse an array of structs where 1 or more elements of the array fail to parse entirely. So we test this exact code snippet over here

**Review by rogerluan — CHANGES_REQUESTED** [review]
Overall logic LGTM 💪 I'd just add 1 extra test case

**Comment by JARMourato** [comment]
It also needs to be linted

File: Sources/Kodable/Codable+Lossless.swift [hunk 2]
```diff
@@ -6,7 +6,7 @@ public typealias LosslessDecodable = LosslessStringConvertible & Decodable
 
 // MARK: Helper type to decode lossless values
 
-struct LosslessValue<T: LosslessDecodable>: Decodable {
+struct LosslessValue<T: Decodable>: Decodable {
```
Comment (JARMourato) [inline Sources/Kodable/Codable+Lossless.swift]: Is this change required ? given this `guard let anyLosslessDecodable = T.self as? LosslessDecodable.Type`

File: Sources/Kodable/Codable+Lossless.swift (hunk 1 above)
Comment (rogerluan) [reply Sources/Kodable/Codable+Lossless.swift]: Edit: nevermind, now I see that `nam` would fail the `name` parsing. I overlooked that haha such similar naming. Perhaps name it after `dragonite` or something, so it's more readable that it would obviously fail to be parsed. Using more descriptive values would also help, e.g. `"dragonite": "this key will fail to be parsed"` 👍 I try to follow those guidelines when writing tests
@matolah

**Review by rogerluan — APPROVED** [review]
💪

**Review by JARMourato — APPROVED** [review]

**Review by JARMourato — APPROVED** [review]
